### PR TITLE
Avoid GtkStock

### PR DESCRIPTION
**Pull request by tamplan**
Add new function to be shared by several modules/applications

### CODE REVIEW

**Comment by tamplan** [comment]
Seen check failed, i search to resolve this...

**Comment by tamplan** [comment]
I forgot to add to track two new files, this is done now but the compile process does not start now...

**Comment by sc0w** [comment]
@mate-desktop/core-team calling to others...

I think it can be on `mate-desktop-utils.c` for one more function

File: libmate-desktop/mate-gtkstock-replacement.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                        const gchar *button_text,
+                        const gchar *icon_name,
+                        gint   response_id)
+{
+    GtkWidget *button;
+
+    button = gtk_button_new_with_mnemonic (button_text);
+
+    // Set button image only if icon_name is set
+    if (icon_name != NULL) {
```
Comment (sc0w) [inline libmate-desktop/mate-gtkstock-replacement.c]: seems you modified the original function with this change

we need this function only to work with icons, if we don't like icons in buttons we have already `gtk_dialog_add_button` , so, we don't need this condition

Comment (tamplan) [reply libmate-desktop/mate-gtkstock-replacement.c]: Yes, i modify the function to test if icon is need or not.
The function call some others functions of GTK in one line:
`    gtk_button_set_use_underline (GTK_BUTTON (button), TRUE);
    gtk_style_context_add_class (gtk_widget_get_style_context (button), "text-button");
    gtk_widget_set_can_default (button, TRUE);
    gtk_widget_show (button);
    gtk_dialog_add_action_widget (GTK_DIALOG (dialog), button, response_id);
`

Comment (tamplan) [reply libmate-desktop/mate-gtkstock-replacement.c]: i have copy/paste header of **mate-gtkstock-replacement.c** from another file, update and giving you credit of this function, feel free to comment here too to advice me.

**Comment by tamplan** [comment]
> @mate-desktop/core-team calling to others...
> 
> I think it can be on `mate-desktop-utils.c` for one more function

Please read here https://github.com/mate-desktop/mate-polkit/pull/52/files, you can see directly than the new file is in charge about only one subject.

Like indicating here https://github.com/mate-desktop/mate-desktop/issues/399#issuecomment-508993566, I'm novice in C programming but I had understood that several files c were finally compiled into one (libmate-desktop), including the correct h file declare the function called from the library.

**Comment by tamplan** [comment]
Maybe planned in future release...

**Pull request closed without merging**